### DISCUSSION TITLE
collection.search querry returns less than  "limit":

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#18204] [Q&A and General discussion] [answered]
When i execute 

```
search_param = {
        "data": [res[0]['embeddings']],
        "anns_field": "embeddings",
        "param": {"metric_type": "L2", "params": {}},
        "limit": 100}
results = collection.search(**search_param)
```

I get less than 100 results.

`["['(distance: 4.3928704424045506e-11, id: 33)', '(distance: 4.3928704424045506e-11, id: 1)', '(distance: 22.99721336364746, id: 28)', '(distance: 618.7476806640625, id: 6)', '(distance: 637.6141967773438, id: 1927712)', '(distance: 642.7174072265625, id: 383706)', '(distance: 645.1341552734375, id: 2477956)', '(distance: 645.2672119140625, id: 1858386)', '(distance: 645.511962890625, id: 2472018)', '(distance: 646.0498046875, id: 1030673)']"]`

But setting limit to 1 has an obvious effect. Only one entry is returned.
Why is that?

**Comment by martinenkoEduard** [comment]
Get it

for hits in results:
    for hit in hits:
        print(f"hit: {hit}, random field:")


But how do could I determine the length of result list?

**Comment by yhmo** [comment, marked as answer]
The results returned by search() is enumerable:
```
    search_param = {
        "data": rng.random((2, dim)),
        "anns_field": vector_field_name,
        "param": {"metric_type": "L2", "params": {"nprobe": 2}},
        "limit": 10,
    }
    results = collection.search(**search_param)
    print("There are", len(results), "groups of results")
    for i, result in enumerate(results):
        print("Search result for {}th vector: ".format(i))
        print("There are", len(result), "pairs of id-distance")
        for j, res in enumerate(result):
            print("Top {}: {}".format(j, res))
```

This script inputs 2 vectors to search, topk is 10, run this script and you will see:

```
There are 2 groups of results
Search result for 0th vector: 
There are 10 pairs of id-distance
Top 0: (distance: 0.0, id: 0)
Top 1: (distance: 13.130629539489746, id: 715)
Top 2: (distance: 13.476598739624023, id: 2290)
Top 3: (distance: 13.549087524414062, id: 2503)
Top 4: (distance: 13.560192108154297, id: 4713)
Top 5: (distance: 13.731561660766602, id: 152)
Top 6: (distance: 13.748649597167969, id: 6206)
Top 7: (distance: 13.893716812133789, id: 2543)
Top 8: (distance: 14.01767635345459, id: 7481)
Top 9: (distance: 14.093172073364258, id: 4485)

Search result for 1th vector: 
There are 10 pairs of id-distance
Top 0: (distance: 0.0, id: 1)
Top 1: (distance: 11.469062805175781, id: 4471)
Top 2: (distance: 12.664854049682617, id: 4492)
Top 3: (distance: 13.094342231750488, id: 4681)
Top 4: (distance: 13.130064010620117, id: 577)
Top 5: (distance: 13.181090354919434, id: 9461)
Top 6: (distance: 13.261234283447266, id: 5506)
Top 7: (distance: 13.344547271728516, id: 237)
Top 8: (distance: 13.386110305786133, id: 4025)
Top 9: (distance: 13.417585372924805, id: 3633)
```

**Reply by martinenkoEduard** [reply]
Thanks! 
Is there a way to a field by its name? 

got it - **res.distance**

**Comment by Yufei447** [comment]
I have the same problem. I set limit equals to 100, but only 17 hits returned. Does anyone know why the case?

**Reply by yhmo** [reply]
Possible reason 1:  there are duplicate primary keys in the collection

Possible reason 2: index is IVF, the "nprobe" value is too small, only a few vectors are searched.
